### PR TITLE
#181414769 feature(provide-feedback):Users should be able to provide feedback

### PR DESCRIPTION
**Pull request by sndicunguye**
**What does this PR do?**

Provide a user feedback

**Description of Task to be completed?**

- Create user feedback route
- A user should be able to provide feedback after spending a day in the accommodation
- Test the route
- Document the route

**How should this be manually tested?**

To test the changes manually:

- clone repo
- switch to `ft-feedback-181414769` branch
- Run `yarn install` to install all dependencies
- Set up `.env` as shown in `.env.example`
- Use **Postman** or any other API Testing platform to make a **POST** request on ... to provide feedback

**What to expect after running the commands:**

A user should be able to give feedback on the accomodation
If the user didn't spend at least one day in the accomodation, he should be able to receive a warning message about the exception

**What are the relevant pivotal tracker stories?**

https://www.pivotaltracker.com/story/show/181414769

**Task implementation plan**

1. Name of task

- Provide User Feedback

2. Pivot Tracker ID

- #181414769

3. Branch Name

- ft-feedback-181414769

4. Description of task (Paraphrase what we have in Pivot Tracker)

Users should be able to comment/provide feedback about an accommodation facility so that "hosts/suppliers" and the company can have better insight as to travellers experience while at their destination(s)
**Condition:** The user should have at least passed one day in the accommodation.

5. Acceptance criteria (Checks to be made)

[ User should be able to provide feedback when he had spent at least a day in the accommodation]
**Scenario**: Given a user registers an accommodation in the Barefoot Nomad, after spending time in the accommodation he can comment out his/her experience in the accommodation

6. Database schema (Migration & models)
 
UserId: INTEGER, accommodationId: INTEGER, feedback: STRING

7. Validations point (Input to validate)

Will use Joi validations to check for any incorrect user input

8. Logic on the access (Explaining who is allowed to CRUD (if applicable) the entities you are working on)

Only users who have spent at least a day in the accommodation will be able to give feedback

9. Logic on actual Implementation

- Setup user feedback model
- Setup the validations using Joi
- Use middleware to check if a user spent at least a day in the accommodation
- Create a user feedback controller
- Create the route for providing feedback
- Test the routes with mocha & chai
- Document the endpoint with swagger

10. Endpoints Format: 

POST  :  Post the user feedback
GET: To view all feedbacks

### CODE REVIEW

File: src/database/models/useraccommodation.js [hunk 1]
```diff
@@ -23,17 +24,16 @@ export default (sequelize, DataTypes) => {
       },
       userId: {
         type: DataTypes.INTEGER,
-        allowNull: false,
       },
       accommodationId: {
         type: DataTypes.INTEGER,
-        allowNull: false,
```
Comment (Robertishimwe) [inline src/database/models/useraccommodation.js]: why did you remove this?

File: src/database/models/useraccommodation.js [hunk 2]
```diff
@@ -23,17 +24,16 @@ export default (sequelize, DataTypes) => {
       },
       userId: {
         type: DataTypes.INTEGER,
-        allowNull: false,
```
Comment (Robertishimwe) [inline src/database/models/useraccommodation.js]: why did you remove this?

File: src/controllers/accommodation.js [hunk 3]
```diff
@@ -184,7 +185,10 @@ class AccommodationController {
       if (!accommodation) {
         return ErrorResponse.notFoundError(res, 'Accommodation not found');
       }
-
+      const userAccommodation = await UserAccommodation.findOne({
+        where: { userId: req.user.id, accommodationId: accommodation.id },
+      });
+      console.log(userAccommodation);
```
Comment (niyojuste) [inline src/controllers/accommodation.js]: What was this added for?

File: src/controllers/accommodation.js [hunk 4]
```diff
@@ -258,6 +262,99 @@ class AccommodationController {
...
+      const findAccommodation = await Accommodation.findOne({
+        where: { id: req.params.id },
+      });
+      if (!findAccommodation) {
+        return ErrorResponse.notFoundError(res, 'Accommodation not found');
+      }
+
+      const deletedFeedback = await UserAccommodation.destroy({
+        where: { accommodationId: req.params.id },
+      });
```
Comment (niyojuste) [inline src/controllers/accommodation.js]: Can you add the userId parameter as well, to delete only the feedback by that user instead of all the feedbacks on the mentioned accommodations

**Review by niyojuste — CHANGES_REQUESTED** [review]

File: src/database/models/useraccommodation.js (hunk 1 above)
Comment (ste3ve3) [reply src/database/models/useraccommodation.js]: My bad, let me add it back

File: src/controllers/accommodation.js (hunk 3 above)
Comment (ste3ve3) [reply src/controllers/accommodation.js]: Forgot to remove it, let me do that really quick

File: src/controllers/accommodation.js (hunk 4 above)
Comment (ste3ve3) [reply src/controllers/accommodation.js]: Yeah, let me fix this too

**Review by niyojuste — APPROVED** [review]

**Review by ericus123 — APPROVED** [review]
